src/composables/useImageUpload.ts

```diff
@@ -0,0 +1,55 @@
+import { ref } from 'vue';
+import { upload as pin } from '@snapshot-labs/pineapple';
+
+export function useImageUpload() {
+  const isUploadingImage = ref(false);
+  const imageUploadError = ref('');
+  const imageUrl = ref('');
+  const imageName = ref('');
+
+  const reset = () => {
+    isUploadingImage.value = false;
+    imageUploadError.value = '';
+    imageUrl.value = '';
+    imageName.value = '';
+  };
+
+  const upload = async (
```

Comment:
What do you think of making it purely async, instead of having callback as well?